host/integration_test.go

```diff
@@ -679,7 +679,7 @@ func (s *integrationSuite) TestWorkflowRetry() {
 		} else {
 			s.Equal(enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_CONTINUED_AS_NEW, events[len(events)-1].GetEventType())
 		}
-		s.Equal(int32(i), events[0].GetWorkflowExecutionStartedEventAttributes().GetAttempt())
+		s.Equal(int32(i+1), events[0].GetWorkflowExecutionStartedEventAttributes().GetAttempt())
```

Comment:
instead can we change the loop to initialize i := 1?

I'm also surprised how this condition is even working:
```
i != maximumAttempts
```